src/Integration/TeamExplorer/SectionController.cs

```diff
@@ -0,0 +1,272 @@
+//-----------------------------------------------------------------------
+// <copyright file="SectionController.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.TeamFoundation.Client.CommandTarget;
+using Microsoft.TeamFoundation.Controls;
+using Microsoft.TeamFoundation.Controls.WPF.TeamExplorer;
+using SonarLint.VisualStudio.Integration.Progress;
+using SonarLint.VisualStudio.Integration.WPF;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Windows.Input;
+
+namespace SonarLint.VisualStudio.Integration.TeamExplorer
+{
+    /// <summary>
+    /// Controller for the SonarLint section in team explorer tool window
+    /// The class is responsible for view and view model creation also hosting the commands
+    /// relevant during the life time of the section (initialized when activated and disposed when navigated to a different section).
+    /// </summary>
+    [TeamExplorerSection(SectionController.SectionId, SonarQubePage.PageId, SectionController.Priority)]
+    internal class SectionController : TeamExplorerSectionBase, IConnectSection
```

Comment:
Aggregates all the commands (connection controller and binding controller provide some in addition to a simple ones provided by this controller directly) which are available whilst the section is not disposed (navigated away).